app/controllers/api/v1/packages_controller.rb

```diff
@@ -129,6 +152,17 @@ def assign_donor_condition?
       def destroy
         @package.really_destroy!
         render json: {}
+      rescue ActiveRecord::InvalidForeignKey
+        raise Goodcity::InventorizedPackageError
+      end
+
+      api :PUT, "/v1/packages/1", "Mark a package as missing"
+      def mark_missing
+        ActiveRecord::Base.transaction do
+          Package::Operations.uninventorize(@package) if PackagesInventory.inventorized?(@package)
+          @package.mark_missing
+        end
+        render json: serializer.new(@package).as_json
```

Comment:
How transaction failing is handled ?

Reply:
Operations raise Goodcity exceptions, which are caught by the controller